packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +96,236 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      // Get the window object from the element's ownerDocument
+      const window = element.ownerDocument?.defaultView;
+      if (!window) {
+        return false;
+      }
+
+      // Get both pathname and search (query parameters)
+      const path = window.location.pathname;
+      const search = window.location.search;
+      const fullUrl = path + search;
+      const regex = parseRegex(argument);
+      return regex.test(fullUrl);
+    } else if (selector.name === 'matches-attr') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const indexOfEqual = argument.indexOf('=');
+      let namePattern: string;
+      let valuePattern: string | undefined;
+      if (indexOfEqual === -1) {
+        namePattern = argument;
+      } else {
+        namePattern = argument.slice(0, indexOfEqual);
+        valuePattern = argument.slice(indexOfEqual + 1);
+      }
+
+      namePattern = stripsWrappingQuotes(namePattern);
+      valuePattern = valuePattern ? stripsWrappingQuotes(valuePattern) : undefined;
+
+      const valueRegex =
+        valuePattern?.startsWith('/') && valuePattern.lastIndexOf('/') > 0
+          ? parseRegex(valuePattern)
+          : undefined;
+
+      if (namePattern.startsWith('/') && namePattern.lastIndexOf('/') > 0) {
+        // matching attribute name by regex
+        const regex = parseRegex(namePattern);
+        const matchingAttrs = [...element.attributes].filter((attr) => regex.test(attr.name));
+
+        // If no value pattern, return true if any attribute matches the name pattern
+        if (!valuePattern) {
+          return matchingAttrs.length > 0;
+        }
+
+        // Check if any of the matching attributes have the specified value
+        return matchingAttrs.some((attr) =>
+          valueRegex ? valueRegex.test(attr.value) : attr.value === valuePattern,
+        );
```

Comment:
IMHO, when finding a match for both key and value, doing it in one pass is more readable than first building the candidate set:

```suggestion
        const regex = parseRegex(namePattern);
        return [...element.attributes].some((attr) => {
          const nameMatches = regex.test(attr.name);
          if (!nameMatches) {
            return false;
          }
          return !valuePattern || (valueRegex ? valueRegex.test(attr.value) : attr.value === valuePattern);
        });
```

Maybe the variables names could be improved (making it a bit harder to read currently). But I left them as is here.